### PR TITLE
download_strategy: replace `bazaar` with `breezy`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Bazaar is no longer maintained, and Breezy seems to be a drop-in
replacement.

I've tested the commands used in the download strategy and they seem to
work.

We need this in order to properly deprecate the `bazaar` formula.

See Homebrew/homebrew-core#106848.

### CODE REVIEW

**Comment by carlocab** [comment]
@bayandin, since I think you looked at this before: this should work, right?

**Review by bayandin — APPROVED** [review]
Yep, this should work

**Comment by bayandin** [comment]
And you probably want to swap `bazaar` with `breezy` here https://github.com/Homebrew/brew/blob/508b48d19e1627db45d33b82e1a2159a010594a5/Library/Homebrew/dependency_collector.rb#L163-L164

**Comment by carlocab** [comment]
Right, thanks. Nice catch.

File: Library/Homebrew/dependency_collector.rb [hunk 1]
```diff
@@ -161,7 +161,7 @@ def resource_dep(spec, tags)
     elsif strategy <= FossilDownloadStrategy
       Dependency.new("fossil", tags)
     elsif strategy <= BazaarDownloadStrategy
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dependency_collector.rb:163]: Should we rename this?

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/dependency_collector.rb (hunk 1 above)
Comment (carlocab) [reply Library/Homebrew/dependency_collector.rb:163]: I think the repository format is still called "Bazaar", so keeping the name is probably still appropriate.

From the Breezy [README](https://github.com/breezy-team/breezy/blob/149cab0ce042ff956f085ea60218ad533d6e1bab/README.rst):

> By default, Breezy provides support for both the [Bazaar](https://www.bazaar-vcs.org/) and [Git](https://www.git-scm.com/) file formats.

Comment (MikeMcQuaid) [reply Library/Homebrew/dependency_collector.rb:163]: Cool, thanks, fine with me then 👍🏻

**Pull request merged**